dashboard/app/models/script.rb

```diff
@@ -121,6 +123,8 @@ def hide_pilot_scripts
       message: 'cannot start with a tilde or dot or contain slashes'
     }
 
+  validates :published_state, acceptance: {accept: SharedConstants::PUBLISHED_STATE.to_h.values.push(nil), message: 'must be nil, in_development, pilot, beta, preview or stable'}
```

Comment:
I don't love that we're constructing this array inline, but I also don't have a specific reason why we shouldn't

Reply:
The nil option will go away when we disallow null. Does that help with your concern?

Reply:
it does!